Task1/Container.cs

```diff
@@ -0,0 +1,132 @@
+                
+                var importProperties = type.GetProperties().Where(prop => prop.GetCustomAttribute<ImportAttribute>() != null);
+                
+                if (importAttribute != null)
+                {
+                    _typeImplementationDependencies.Add(type, type);
+                }
+                
+                else if (exportAttribute != null)
+                {
+                    _typeImplementationDependencies.Add(
+                        exportAttribute.Contract == null ? type : exportAttribute.Contract, type);
+                }
+                
+                else if(importProperties.Any())
+                {
+                    _typeImplementationDependencies.Add(type, type);
+                }
+            }
+        }
+
+        public void AddType(Type type)
+        {
+            _typeImplementationDependencies.Add(type, type);
+        }
+
+        public void AddType(Type type, Type baseType)
+        {
+            _typeImplementationDependencies.Add(baseType, type);
+        }
+
+        public T Get<T>()
+        {
+            var implementationType =
+                _typeImplementationDependencies.LastOrDefault(serviceDescriptor => serviceDescriptor.Key == typeof(T) || serviceDescriptor.Value == typeof(T)).Value;
+
+            if (implementationType == null)
+            {
+                throw new Exception($"The implementation for the {typeof(T)} wasn't found");
+            }
+            
+            var constructorParameters = implementationType.GetConstructors()[0].GetParameters();
+            
+            var importedProperties = implementationType.GetProperties()
+                .Where(prop => prop.GetCustomAttribute<ImportAttribute>() != null).ToList();
+            
+            if (constructorParameters.Length > 0)
+            {
+                return GetImplementationForTypeWithConstructor<T>(implementationType, constructorParameters);
+            }
+            
+            if (importedProperties.Any())
+            {
+                return GetImplementationForTypeWithImportedProperties<T>(implementationType, importedProperties);
+            }
+
+            return (T) Activator.CreateInstance(implementationType);
+        }
+
+        private T GetImplementationForTypeWithImportedProperties<T>(Type implementationType, IEnumerable<PropertyInfo> propertyInfos)
+        {
+            var implementation = Activator.CreateInstance(implementationType);
+            
+            foreach (var property in implementationType.GetProperties().Where(prop => prop.GetCustomAttribute<ImportAttribute>() != null))
+            {
+                var parameterType = property.PropertyType;
+                
+                var method = typeof(Container).GetMethod(nameof(Get));
+                
+                var generic = method.MakeGenericMethod(parameterType);
+                
+                object propertyImplementation = null;
+                
```

Comment:
Remove empty line after line - use empty lines to organize code in logical blocks.